src/charm.py

```diff
@@ -164,8 +164,10 @@ def run_pre_refresh_checks_after_1_unit_refreshed() -> None:
         pass
 
     def run_pre_refresh_checks_before_any_units_refreshed(self) -> None:
-        if not self._charm._patroni.are_all_members_ready():
-            raise charm_refresh.PrecheckFailed("PostgreSQL is not running on 1+ units")
+        for attempt in Retrying(stop=stop_after_attempt(2), wait=wait_fixed(1), reraise=True):
+            with attempt:
+                if not self._charm._patroni.are_all_members_ready():
+                    raise charm_refresh.PrecheckFailed("PostgreSQL is not running on 1+ units")
```

Comment:
When moving the standby leader, the status becomes stopped for a bit, so the action will fail if the standby leader is not already the lowest unit.